storj/downloader.py

```diff
@@ -193,30 +174,37 @@ def create_download_connection(self, url, path_to_save, shard_index):
             farmer_tries += 1
 
             try:
-                r = requests.get(url)
-                # Write the file
-                with open(path_to_save, 'wb') as f:
-                    for chunk in r.iter_content(chunk_size=1024):
-                        if chunk:  # filter out keep-alive new chunks
-                            f.write(chunk)
+                # data is spooled in memory until the file size exceeds max_size
+                shard = SpooledTemporaryFile(self.max_spooled, 'wb')
+
+                # Request the shard
+                r = requests.get(url, stream=True)
                 if r.status_code != 200 and r.status_code != 304:
                     raise StorjFarmerError()
 
+                # Write the file
+                for chunk in r.iter_content(chunk_size=1024):
+                    if chunk:  # filter out keep-alive new chunks
+                        shard.write(chunk)
+
+                # Everything ok
+                # move file read pointer at beginning
+                shard.seek(0)
+                return shard
+
             except StorjFarmerError as e:
                 self.__logger.error(e)
                 self.__logger.error("First try failed. Retrying... (%s)" %
                                     str(farmer_tries))  # update shard download state
-                continue
 
             except Exception as e:
-                self.__logger.error("Unhandled error")
-                self.__logger.error("Error occured while downloading shard \
-                    at index %s. Retrying... (%s)" % (shard_index, farmer_tries)
-                                    )
                 self.__logger.error(e)
-                continue
-            else:
-                break
+                self.__logger.error("Unhandled error")
+                self.__logger.error("Error occured while downloading shard at "
+                                    "index %s. Retrying... (%s)" % (shard_index, farmer_tries))
```

Comment:
no need to `%` you can just use `"index %s. Retrying... (%s)", shard_index, farmer_tries)`.
the `logging` module will do the formatting for you.